pkg/ipcache/metadata.go

```diff
@@ -427,18 +427,28 @@ func (ipc *IPCache) resolveIdentity(ctx context.Context, prefix netip.Prefix, in
 	}
 
 	lbls := info.ToLabels()
-	if (lbls.Has(labels.LabelWorld[labels.IDNameWorld]) ||
-		lbls.Has(labels.LabelWorldIPv4[labels.IDNameWorldIPv4]) ||
-		lbls.Has(labels.LabelWorldIPv6[labels.IDNameWorldIPv6])) &&
-		(lbls.Has(labels.LabelRemoteNode[labels.IDNameRemoteNode]) ||
-			lbls.Has(labels.LabelHost[labels.IDNameHost])) {
-		// If the prefix is associated with both world and (remote-node or
-		// host), then the latter (remote-node or host) take precedence to
-		// avoid allocating a CIDR identity for an entity within the cluster.
+
+	// If we are restoring an identity in the remote-node identity scope, we need to unconditionally add remote-node and cidr labels back
+	if restoredIdentity.Scope() == identity.IdentityScopeRemoteNode {
```

Comment:
Couldn't we short-circuit this if we check for the feature flag so that we avoid adding CIDR labels here? I'm thinking in the flag=disabled case, the restored identity will be have the remote node scope, but the feature is off so we shouldn't be adding CIDR labels to begin win.

Reply:
Related: What happens if there's a restored identity and the flag is off, then this function whittles down the labels to end up with `remote-node`. Will the identity allocator below just end up ignoring the request to restore this specific identity number and return the `reserved:remote-node` identity `6`?

Reply:
> Couldn't we short-circuit this if we check for the feature flag so that we avoid adding CIDR labels here?

Sure. easy.

> Related: What happens if there's a restored identity and the flag is off...?

Then we allocate the correct identity; `AllocateIdentity()` unconditionally calls `LookupReservedIdentityByLabels()` first, which does the right thing.

Reply:
Either way, this is moot; we wont try and restore this if the flag is off.

Reply:
Could you add a code comment as to why it's moot to check the feature flag? It seems like something someone will stubble across and prevent them from having to understand the rest implementation to come to the same realization.

Reply:
@nathanjsweet This is unresolved ^

Reply:
@christarazi I added a comment several lines down where the CIDR is added.

Reply:
I meant that the code comment should explain why it's not necessary to additionally check the feature flag, as the original comment on this line suggested.